logbus/logstreamer/logstreamer.go

```diff
@@ -124,40 +218,66 @@ func (ls *LogStreamer) tryStream(ctx context.Context) (bool, error) {
 	return false, nil
 }
 
-// Write writes the given delta to the log streamer.
+// WriteAsync queues delta up for writing and returns immediately. The returned
+// channel will be closed when delta has been passed off to the buffer.
+//
+// This is different than Write in two important ways:
+//
+// 1. It will never block.
+//
+// 2. If the buffer is full, it cannot guarantee order the order that delta will
+// be added to the buffer in the order in which WriteAsync was called.
```

Comment:
nit: This comment reads a little oddly
```suggestion
// 2. If the buffer is full, it cannot guarantee the order that delta will
// be added to the buffer matches the order in which WriteAsync was called.
```